src/batou_ext/python.py

```diff
@@ -217,26 +218,45 @@ def update(self):
                 return
 
             # add user env to DT_RPATH
-            self.__patchelf(["--add-rpath", directories], files_to_fix)
-            # drop everything from DT_RPATH except self.env_directory
+            # & drop everything from DT_RPATH except self.env_directory
             # and directories in the venv (to allow shared libraries from numpy
             # to load other shared libraries from numpy).
             self.__patchelf(
                 [
-                    "--shrink-rpath",
+                    "--add-rpath-and-shrink",
+                    directories,
                     "--allowed-rpath-prefixes",
                     f"{directories}:$ORIGIN",
                 ],
                 files_to_fix,
             )
 
     def __patchelf(self, args, paths):
+        patchelf_expr = shlex.quote(
+            dedent(
+                """
+        with import <nixpkgs> {}; {
+          patchelf = pkgs.patchelf-venv or patchelf.overrideAttrs ({ patches ? [], ... }: {
+            patches = patches ++ [
+              (fetchpatch {
+                url = "https://github.com/flyingcircusio/patchelf/commit/6ffde887d77275323c81c2e091891251b021abb3.patch";
```

Comment:
To the person reviewing this, please also take a look at the patchelf patch itself and give feedback if needed. I'd like to throw this against upstream at some point.

Reply:
I cannot really say anything useful about the patch 🤷

Reply:
@zagy would you agree that the approach I took here is reasonable? I.e. doing both operations in a single operation now and leaving the batou component as-is?

Reply:
Yeah, well. If it's stupid but works it ain't stupid.